plugins/mumble_plugin_linux.h

```diff
@@ -193,22 +193,30 @@ bool peekProc(PTR_TYPE base, T &dest) {
 	return (nread != -1 && static_cast<size_t>(nread) == in.iov_len);
 }
 
-static bool inline initialize(const std::multimap<std::wstring, unsigned long long int> &pids, const wchar_t *procname, const wchar_t *modname = NULL) {
+static bool inline initialize(MumblePIDLookup lookupFunc, MumblePIDLookupContext lookupContext, const wchar_t *procname, const wchar_t *modname = NULL) {
 	pModule = 0;
 
-	if (! pids.empty()) {
-		std::multimap<std::wstring, unsigned long long int>::const_iterator iter = pids.find(std::wstring(procname));
-
-		if (iter != pids.end())
-			pPid = static_cast<pid_t>(iter->second);
-		else
-			pPid = 0;
-	} else {
-		pPid = 0;
-	}
-
-	if (pPid == 0)
-		return false;
+        if (lookupFunc != NULL && lookupContext != NULL) {
+                while (1) {
```

Comment:
I don't like this. As status is a plain integer it is not obvious this isn't an infinite loop which kinda suggests to me we shouldn't write it this way.

Reply:
```cpp
for(;;)
```
is obviously the way to go.

I'm not sure this implementation was in its final state mkrautz intended.